### PR TITLE
Setup Wizard: Add Creative Mail Recommendation

### PR DESCRIPTION
**Pull request by robertf4**
#### Changes proposed in this Pull Request:

This PR adds a recommendation to the Setup Wizard for Creative Mail.

When Creative Mail is not installed:
image

image

#### Jetpack product discussion

pbtFPC-Hb-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

1. On your Jetpack site add the following to then end of the wp-config.php file in order to make the Setup Wizard show:
```
add_filter( 'jetpack_show_setup_wizard', '__return_true' );
add_filter( 'jetpack_pre_connection_prompt_helpers', '__return_true' );
```
2. Visit `/wp-admin/admin.php?page=jetpack#/setup/features` and scroll to the Marketing section.
3. Verify that the Creative Mail recommendation shows with a disabled toggle and an install button.
4. Click "Install Now" and verify that the Creative Mail plugin gets installed and activated.
5. Click "Configure" and verify that you are taken to the Creative Mail setup flow.
6. Click the "Learn more" link and verify that it navigates to a jetpack support page. (Note: this page does not yet exist but will before release)
7. Deactivate the Creative Mail plugin but leave it installed. Go to the recommendation in the Setup Wizard and click "Install Now". Verify that the Creative Mail plugin is activated.

#### Proposed changelog entry for your changes:

* Added Creative Mail recommendation to the Setup Wizard.

### CODE REVIEW

**Comment by robertf4** [comment]
> It worked well. Would it be possible to add an indicator after clicking install to indicate that the plugin is installing?
> 
> The first time I tested, it took just long enough that I was doubting if I clicked it or if it was working.

Good call, I really should have added one to begin with. I've added a spinner in the middle of the install button during install.

File: _inc/client/setup-wizard/feature-toggle/map-feature-to-props.js [hunk 1]
```diff
@@ -293,6 +295,39 @@ const features = {
...
+				'creative-mail-by-constant-contact/creative-mail-plugin.php'
+			);
+
+			return {
+				feature: 'creative-mail',
+				title: __( 'Creative Mail by Constant Contact', 'jetpack' ),
+				details: __( 'Turn visitors into subscribers with email marketing.', 'jetpack' ),
+				checked: isCreativeMailActive,
+				isDisabled: true,
+				isPaid: true,
```
Comment (jeherve) [inline _inc/client/setup-wizard/feature-toggle/map-feature-to-props.js:311]: Do we need this here? The plugin is free.

File: _inc/client/setup-wizard/feature-toggle/map-feature-to-props.js [hunk 2]
```diff
@@ -293,6 +295,39 @@ const features = {
...
+				configureLink: isCreativeMailActive ? '/wp-admin/admin.php?page=creativemail' : null,
+				learnMoreLink:
+					'https://jetpack.com/support/jetpack-blocks/subscriber-form-block-and-field/',
```
Comment (jeherve) [inline _inc/client/setup-wizard/feature-toggle/map-feature-to-props.js:314]: I've updated this link to be a bit more descriptive.

```suggestion
					'https://jetpack.com/support/jetpack-blocks/form-block/newsletter-sign-up-form/',
```

**Review by jeherve — CHANGES_REQUESTED** [review]
This tests well. I only have 2 remarks.

**Review by jeherve — APPROVED** [review]

**Pull request merged**